tests/utils/test_augmentation.py

```diff
@@ -0,0 +1,251 @@
+                "Frac2": [0.9, 0.4],
+                "Other1": ["A", "B"],
+                "Other2": ["C", "D"],
+            },
+            [["Slot1", "Frac1"], ["Slot2", "Frac2"]],
+            {
+                "Slot1": ["s1", "s2", "s2", "s4"],
+                "Slot2": ["s2", "s4", "s1", "s2"],
+                "Frac1": [0.1, 0.6, 0.9, 0.4],
+                "Frac2": [0.9, 0.4, 0.1, 0.6],
+                "Other1": ["A", "B", "A", "B"],
+                "Other2": ["C", "D", "C", "D"],
+            },
+            id="2inv_2dependent_2add",
+        ),
+        param(  # 2 invariant cols, 2 dependent ones, 2 additional ones
+            {
+                "Slot1": ["s1", "s2"],
+                "Slot2": ["s2", "s4"],
+                "Frac1": [0.1, 0.6],
+                "Frac2": [0.9, 0.4],
+                "Temp1": [10, 20],
+                "Temp2": [50, 60],
+                "Other": ["x", "y"],
+            },
+            [["Slot1", "Frac1", "Temp1"], ["Slot2", "Frac2", "Temp2"]],
+            {
+                "Slot1": ["s1", "s2", "s2", "s4"],
+                "Slot2": ["s2", "s4", "s1", "s2"],
+                "Frac1": [0.1, 0.6, 0.9, 0.4],
+                "Frac2": [0.9, 0.4, 0.1, 0.6],
+                "Temp1": [10, 20, 50, 60],
+                "Temp2": [50, 60, 10, 20],
+                "Other": ["x", "y", "x", "y"],
+            },
+            id="2inv_4dependent2each_1add",
+        ),
+    ],
+)
+def test_df_permutation_aug(data, col_groups, data_expected):
+    """Test permutation invariance data augmentation is done correctly."""
+    # Create all needed dataframes
+    df = pd.DataFrame(data)
+    df_augmented = df_apply_permutation_augmentation(df, col_groups)
+    df_expected = pd.DataFrame(data_expected)
+
+    # Determine equality ignoring row order
+    are_equal = (
+        pd.merge(left=df_augmented, right=df_expected, how="outer", indicator=True)[
+            "_merge"
+        ]
+        .eq("both")
+        .all()
+    )
+
+    assert (
+        are_equal
+    ), f"\norig:\n{df}\n\naugmented:\n{df_augmented}\n\nexpected:\n{df_expected}"
+
+
+@pytest.mark.parametrize(
+    ("col_groups", "msg"),
+    [
+        param([], "at least two column sequences", id="no_groups"),
+        param([["A"]], "at least two column sequences", id="just_one_group"),
+        param([["A"], ["B", "C"]], "the amount of columns in", id="different_lengths"),
+        param([[], []], "each column group has", id="empty_group"),
+    ],
+)
+def test_df_permutation_aug_invalid(col_groups, msg):
+    """Test correct errors for invalid permutation attempts."""
+    df = pd.DataFrame({"A": [1, 1], "B": [2, 2], "C": ["x", "y"]})
+    with pytest.raises(ValueError, match=msg):
+        df_apply_permutation_augmentation(df, col_groups)
+
+
+@pytest.mark.parametrize(
+    ("data", "causing", "affected", "data_expected"),
+    [
+        param(  # 1 causing val, 1 col affected (with 3 values)
+            {
+                "A": [0, 1],
```

Comment:
we do not do anything to duplicates at all because augmentation should augment duplciates as they are to not change the relative weight between the points
and it does in cits urrent implementation:
![image](https://github.com/emdgroup/baybe/assets/17951239/3750588d-e716-48ec-bf05-9996fae11a46)

Reply:
Ok, looking good, thx 👌🏼